docs/src/proposals/scrambled-tx.md

```diff
@@ -0,0 +1,335 @@
+the recursive sha256 is called *descrambling*.
+
+### Summary of consensus changes
+
+In addition to the obvious change of submission of scrambled transactions by
+clients, this proposal introduces a new pipeline stage (called the descrambling
+stage) and splits the voting into 2 phases (pre-vote and vote).
+
+Firstly, the leader feeds incoming transactions down the turbine immediately
+with minimum sanitization (This assumes leaders are now so-called bank-less and
+scheduling is delayed and on-chain via mempool).
+
+The validators pre-vote on the completed block, once they recover all data.
+Thus, the opt-conf commitment of data availability of the block can be rather
+established rather quickly, without replay timing variation by all other
+validators (This average duration will be the basis of difficulty adjustment of
+scrambling).
+
+Also, rpc can notify optimistic execution signal to clients at this point. This
+signifies subsequent pipeline processing is fully deterministic and not reverted
+unless opt-conf is violated (some portion of stake is slash-able).
+
+At the same time, the validators also initiate the descrambling stage for the
+block in a random, collaborative and distributed manner. Once they solve their
+chosen chunk of scrambled transactions, they send revealed encryption keys to
+the leader to be propagated via turbine likewise the block shreds. Also
+sig-verify is done at the end of descrambling.
+
+After all transactions are descrambled from the cooperative efforts, the block
+can be finally scheduled and executed deterministically.
+
+Lastly, validators votes on the reproduced bank hash after replaying the
+processing likewise. Then, rpc can notify optimistic confirmation signal to
+clients.
+
+### Downsides
+
+- Increased latency
+- Some computational cost on clients
+- Reliance of local wall clock time
+- Pretty invasive system change
+
+
+### Assumptions of other subsystem's improvements
+
+- bankless leader
+- randomized leader schedule (or stop 4-slot grouping at the very least)
+- on-chain mempool with bidding
+- quic adaption for tpu (for ipv6-mtu-exeeding transaction packet and ddos prevention)
+
+### Scrambling details
+
+As said above, scrambling refers to a specially-arranged encryption scheme where
+anyone can decrypt (= reveal) the transaction details after a non-parallelizable
+computation which takes certain amount of time.
+
+On top of it, scrambling also has a nice property where the revelation can
+easily be verified by anyone with some kind of message authentication tag. This
+means rpc node doesn't need GPU.
+
+All of fields in a transaction are scrambled with randomized padding, including
+fee payer to avoid any kind of fingerprinting while allowing garbage-based
+spamming with amplification risk.
+
+Also, scrambled transaction expires very quickly based on wall clock time, less
+than the targeted duration of difficulty to protect from processing stalls by
+malicious leaders for MEV.
```

Comment:
This only works if there is a significant incentive for miners to reveal their hash power [like I outlined above](https://github.com/solana-labs/solana/pull/23837/files#r1306645585), right ?

Btw, does this work with [durable nonces](https://docs.solana.com/offline-signing/durable-nonce) ? I think those durable nonce transactions can't expire nearly as fast.

Reply:
> This only works if there is a significant incentive for miners to reveal their hash power [like I outlined above](https://github.com/solana-labs/solana/pull/23837/files#r1306645585), right ?

as i relied here (https://github.com/solana-labs/solana/pull/23837#discussion_r1335136208), pow-like hash power isn't irrelevant for descrambeling. fastest de-scrambling speed can be assumed of public knowledge.

Reply:
> Btw, does this work with [durable nonces](https://docs.solana.com/offline-signing/durable-nonce) ? I think those durable nonce transactions can't expire nearly as fast.

it's technically possible, but considering the intention of nonce for off-line signing, nonced tx won't be scrambled usually.

Reply:
looks like my understanding was wrong, as I [mentioned above](https://github.com/solana-labs/solana/pull/23837#discussion_r1347120491)